lightning-invoice/src/lib.rs

```diff
@@ -901,6 +906,79 @@ impl Deref for PositiveTimestamp {
 }
 
 impl Invoice {
+	/// Utility to construct an invoice.
+	pub fn from_channelmanager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
+		channelmanager: &ChannelManager<Signer, M, T, K, F, L>, amt_msat: Option<u64>, description: String,
+		network: Network,
+	) -> Result<Invoice, CreationError>
```

Comment:
Could you move this into a separate utility module akin to `lightning_block_sync::init`? I'd prefer to separate higher-level lightning data structures from lower-level invoice code when possible.

Reply:
Hm, what do you think the module should be called? `utils` perhaps?

Reply:
Yeah, `utils` would be fine.